core/src/main/java/org/infinispan/reactive/publisher/impl/ClusterPublisherManagerImpl.java

```diff
@@ -1042,8 +1024,7 @@ void completeSegment(int segment) {
        * via {@link Subscriber#onError(Throwable)}.
        */
       boolean handleThrowable(Throwable t, Address target, IntSet segments) {
-         // Most likely SuspectException will be wrapped in CompletionException
-         if (t instanceof SuspectException || t.getCause() instanceof SuspectException) {
+         if (t instanceof SuspectException) {
```

Comment:
Did you find the only wrapping is a CompletionException? Which I see is handled at `InnerPublisherSubscription#handleThrowInResponse` invocation.

Reply:
Yes, there's nothing else wrapping SuspectException